plugins/aws_sts/lib/samson_aws_sts/samson_plugin.rb

```diff
@@ -0,0 +1,38 @@
+module SamsonAwsSts
+  ROLE_SESSION_NAME = 'StaticAssetsPipeline'
+
+  class Engine < Rails::Engine
+  end
+
+  class << self
+    def set_env_vars(sts_client:, stage:)
+      creds = sts_client.assume_role(
+        role_arn: stage.aws_sts_iam_role_arn,
+        role_session_name: ROLE_SESSION_NAME
+      ).credentials
+
+      ENV['STS_AWS_ACCESS_KEY_ID']     = creds.access_key_id
+      ENV['STS_AWS_SECRET_ACCESS_KEY'] = creds.secret_access_key
+      ENV['STS_AWS_SESSION_TOKEN']     = creds.session_token
```

Comment:
should not set global env vars ... they will never expire and be shared between all deploys/stagges